File: src/test/resources/verticle/assertions.js
var TestSuite = require('vertx-unit-js/test_suite');
var suite = TestSuite.create("the_suite");

suite.test("assert_equals", function(context) {
    context.assertEquals(0, 0);
    context.assertEquals(10, 10);
    context.assertEquals(0.12, 0.12);
    context.assertEquals(true, true);
    context.assertEquals(false, false);
    context.assertEquals("the_string", "the_" + "string");
    context.assertEquals({foo:1,bar:"juu"}, {foo:1,bar:"juu"});
    context.assertEquals([], []);
    context.assertEquals([123], [123]);
}).test("fail_assert_equals_1", function(context) {
    context.assertEquals(0, 1);
}).test("fail_assert_equals_2", function(context) {
    context.assertEquals(0.12, 0.13);
}).test("fail_assert_equals_3", function(context) {
    context.assertEquals(true, false);
}).test("fail_assert_equals_4", function(context) {
    context.assertEquals(false, true);
}).test("fail_assert_equals_5", function(context) {
    context.assertEquals("foo", "bar");
}).test("fail_assert_equals_6", function(context) {
    context.assertEquals({foo:1,bar:"juu"}, {foo:1,bar:"daa"});
}).test("fail_assert_equals_7", function(context) {
    context.assertEquals([123], [321]);
});

suite.test("assert_in_range", function(context) {
    context.assertInRange(0.12, 0.13, 0.02);
    context.assertInRange(0.12, 0.11, 0.02);
}).test("fail_assert_in_range_1", function(context) {
    context.assertInRange(0.12, 0.15, 0.02);
}).test("fail_assert_in_range_2", function(context) {
    context.assertInRange(0.12, 0.09, 0.02);
});

suite.test("assert_equals_with_message", function(context) {
    context.assertEquals(0, 0, "the_message");
    context.assertEquals(10, 10, "the_message");
    context.assertEquals(0.12, 0.12, "the_message");
    context.assertEquals(true, true, "the_message");
    context.assertEquals(false, false, "the_message");
    context.assertEquals("the_string", "the_" + "string", "the_message");
    context.assertEquals({foo:1,bar:"juu"}, {foo:1,bar:"juu"}, "the_message");
    context.assertEquals([], [], "the_message");
    context.assertEquals([123], [123], "the_message");
}).test("fail_assert_equals_with_message_1", function(context) {
    context.assertEquals(0, 1, "the_message");
}).test("fail_assert_equals_with_message_2", function(context) {
    context.assertEquals(0.12, 0.13, "the_message");
}).test("fail_assert_equals_with_message_3", function(context) {
    context.assertEquals(true, false, "the_message");
}).test("fail_assert_equals_with_message_4", function(context) {
    context.assertEquals(false, true, "the_message");
}).test("fail_assert_equals_with_message_5", function(context) {
    context.assertEquals("foo", "bar", "the_message");
}).test("fail_assert_equals_with_message_6", function(context) {
    context.assertEquals({foo:1,bar:"juu"}, {foo:1,bar:"daa"}, "the_message");
}).test("fail_assert_equals_with_message_7", function(context) {
    context.assertEquals([123], [321], "the_message");
});

suite.test("assert_not_equals", function(context) {
    context.assertNotEquals(0, 1);
    context.assertNotEquals(0.12, 0.13);
    context.assertNotEquals(true, false);
    context.assertNotEquals(false, true);
    context.assertNotEquals("foo", "bar");
    context.assertNotEquals({foo:1,bar:"juu"}, {foo:1,bar:"daa"});
    context.assertNotEquals([123], [321]);
}).test("fail_assert_equals_1", function(context) {
    context.assertNotEquals(0, 0);
}).test("fail_assert_equals_2", function(context) {
    context.assertNotEquals(0.12, 0.12);
}).test("fail_assert_equals_3", function(context) {
    context.assertNotEquals(true, true);
}).test("fail_assert_equals_4", function(context) {
    context.assertNotEquals(false, false);
}).test("fail_assert_equals_5", function(context) {
    context.assertNotEquals("the_string", "the_" + "string");
}).test("fail_assert_equals_6", function(context) {
    context.assertNotEquals({foo:1,bar:"juu"}, {foo:1,bar:"juu"});
}).test("fail_assert_equals_7", function(context) {
    context.assertNotEquals([], []);
}).test("fail_assert_equals_8", function(context) {
    context.assertNotEquals([123], [123]);
});

suite.test("assert_not_equals_with_message", function(context) {
    context.assertNotEquals(0, 1, "the_message");
    context.assertNotEquals(0.12, 0.13, "the_message");
    context.assertNotEquals(true, false, "the_message");
    context.assertNotEquals(false, true, "the_message");
    context.assertNotEquals("foo", "bar", "the_message");
    context.assertNotEquals({foo:1,bar:"juu"}, {foo:1,bar:"daa"}, "the_message");
    context.assertNotEquals([123], [321], "the_message");
}).test("fail_assert_equals_with_message_1", function(context) {
    context.assertNotEquals(0, 0, "the_message");
}).test("fail_assert_equals_with_message_2", function(context) {
    context.assertNotEquals(0.12, 0.12, "the_message");
}).test("fail_assert_equals_with_message_3", function(context) {
    context.assertNotEquals(true, true, "the_message");
}).test("fail_assert_equals_with_message_4", function(context) {
    context.assertNotEquals(false, false, "the_message");
}).test("fail_assert_equals_with_message_5", function(context) {
    context.assertNotEquals("the_string", "the_" + "string", "the_message");
}).test("fail_assert_equals_with_message_6", function(context) {
    context.assertNotEquals({foo:1,bar:"juu"}, {foo:1,bar:"juu"}, "the_message");
}).test("fail_assert_equals_with_message_7", function(context) {
    context.assertNotEquals([], [], "the_message");
}).test("fail_assert_equals_with_message_8", function(context) {
    context.assertNotEquals([123], [123], "the_message");
});

suite.test("assert_true", function(context) {
    context.assertTrue(true);
}).test("fail_assert_true", function(context) {
    context.assertTrue(false);
});

suite.test("assert_true_with_message", function(context) {
    context.assertTrue(true, "the_message");
}).test("fail_assert_true_with_message", function(context) {
    context.assertTrue(false, "the_message");
});

suite.test("assert_false", function(context) {
    context.assertFalse(false);
}).test("fail_assert_false", function(context) {
    context.assertFalse(true);
});

suite.test("assert_false_with_message", function(context) {
    context.assertFalse(false, "the_message");
}).test("fail_assert_false_with_message", function(context) {
    context.assertFalse(true, "the_message");
});

suite.test("fail_", function(context) {
    context.fail();
});

suite.test("fail_with_message", function(context) {
    context.fail("the_message");
});

suite.run(vertx, {reporters:[{to: "bus", at: "assert_tests"}]});

File: src/test/resources/verticle/assertions.rb
require 'vertx-unit/test_suite'
suite = VertxUnit::TestSuite.create 'the_suite'


suite.test 'assert_null' do |context| context.assert_null(nil) end
suite.test 'fail_assert_null_1' do |context| context.assert_null(0) end
suite.test 'fail_assert_null_2' do |context| context.assert_null('string') end
suite.test 'fail_assert_null_3' do |context| context.assert_null(0.12) end
suite.test 'fail_assert_null_4' do |context| context.assert_null(true) end
suite.test 'fail_assert_null_5' do |context| context.assert_null(false) end
suite.test 'fail_assert_null_6' do |context| context.assert_null({}) end
suite.test 'fail_assert_null_7' do |context| context.assert_null([]) end

suite.test 'assert_null_with_message' do |context| context.assert_null(nil, 'the_message') end
suite.test 'fail_assert_null_with_message_1' do |context| context.assert_null(0, 'the_message') end
suite.test 'fail_assert_null_with_message_2' do |context| context.assert_null('string', 'the_message') end
suite.test 'fail_assert_null_with_message_3' do |context| context.assert_null(0.12, 'the_message') end
suite.test 'fail_assert_null_with_message_4' do |context| context.assert_null(true, 'the_message') end
suite.test 'fail_assert_null_with_message_5' do |context| context.assert_null(false, 'the_message') end
suite.test 'fail_assert_null_with_message_6' do |context| context.assert_null({}, 'the_message') end
suite.test 'fail_assert_null_with_message_7' do |context| context.assert_null([], 'the_message') end

suite.test 'assert_not_null' do |context|
  context.assert_not_null(0)
  context.assert_not_null('string')
  context.assert_not_null(0.12)
  context.assert_not_null(true)
  context.assert_not_null(false)
  context.assert_not_null({})
  context.assert_not_null([])
end
suite.test 'fail_assert_null' do |context|
  context.assert_not_null(nil)
end

suite.test 'assert_not_null_with_message' do |context|
  context.assert_not_null(0, 'the_message')
  context.assert_not_null('string', 'the_message')
  context.assert_not_null(0.12, 'the_message')
  context.assert_not_null(true, 'the_message')
  context.assert_not_null(false, 'the_message')
  context.assert_not_null({}, 'the_message')
  context.assert_not_null([], 'the_message')
end
suite.test 'fail_assert_null_with_message' do |context|
  context.assert_not_null(nil, 'the_message')
end

suite.test 'assert_equals' do |context|
  context.assert_equals(0, 0)
  context.assert_equals(10, 10)
  context.assert_equals(0.12, 0.12)
  context.assert_equals(true, true)
  context.assert_equals(false, false)
  context.assert_equals('the_string', 'the_' + 'string')
  context.assert_equals({:foo=>1,:bar=>'juu'}, {:foo=>1,:bar=>'juu'})
  context.assert_equals([], [])
  context.assert_equals([123], [123])
end
suite.test 'fail_assert_equals_1' do |context| context.assert_equals(0, 1) end
suite.test 'fail_assert_equals_2' do |context| context.assert_equals(0.12, 0.13) end
suite.test 'fail_assert_equals_3' do |context| context.assert_equals(true, false) end
suite.test 'fail_assert_equals_4' do |context| context.assert_equals(false, true) end
suite.test 'fail_assert_equals_5' do |context| context.assert_equals('foo', 'bar') end
suite.test 'fail_assert_equals_6' do |context| context.assert_equals({:foo=>1,:bar=>'juu'}, {:foo=>1,:bar=>'daa'}) end
suite.test 'fail_assert_equals_7' do |context| context.assert_equals([123], [321]) end

suite.test 'assert_in_range' do |context|
  context.assert_in_range(0.12, 0.13, 0.02)
  context.assert_in_range(0.12, 0.11, 0.02)
end
suite.test 'fail_assert_in_range_1' do |context|
  context.assert_in_range(0.12, 0.15, 0.02)
end
suite.test 'fail_assert_in_range_2' do |context|
  context.assert_in_range(0.12, 0.09, 0.02)
end

suite.test 'assert_equals_with_message' do |context|
  context.assert_equals(0, 0, 'the_message')
  context.assert_equals(10, 10, 'the_message')
  context.assert_equals(0.12, 0.12, 'the_message')
  context.assert_equals(true, true, 'the_message')
  context.assert_equals(false, false, 'the_message')
  context.assert_equals('the_string', 'the_' + 'string', 'the_message')
  context.assert_equals({:foo=>1,:bar=>'juu'}, {:foo=>1,:bar=>'juu'}, 'the_message')
  context.assert_equals([], [], 'the_message')
  context.assert_equals([123], [123], 'the_message')
end
suite.test 'fail_assert_equals_with_message_1' do |context| context.assert_equals(0, 1, 'the_message') end
suite.test 'fail_assert_equals_with_message_2' do |context| context.assert_equals(0.12, 0.13, 'the_message') end
suite.test 'fail_assert_equals_with_message_3' do |context| context.assert_equals(true, false, 'the_message') end
suite.test 'fail_assert_equals_with_message_4' do |context| context.assert_equals(false, true, 'the_message') end
suite.test 'fail_assert_equals_with_message_5' do |context| context.assert_equals('foo', 'bar', 'the_message') end
suite.test 'fail_assert_equals_with_message_6' do |context| context.assert_equals({:foo=>1,:bar=>'juu'}, {:foo=>1,:bar=>'daa'}, 'the_message') end
suite.test 'fail_assert_equals_with_message_7' do |context| context.assert_equals([123], [321], 'the_message') end

suite.test 'assert_not_equals' do |context|
  context.assert_not_equals(0, 1)
  context.assert_not_equals(0.12, 0.13)
  context.assert_not_equals(true, false)
  context.assert_not_equals(false, true)
  context.assert_not_equals('foo', 'bar')
  context.assert_not_equals({:foo=>1,:bar=>'juu'}, {:foo=>1,:bar=>'daa'})
  context.assert_not_equals([123], [321])
end
suite.test 'fail_assert_equals_1' do |context| context.assert_not_equals(0, 0)  end
suite.test 'fail_assert_equals_2' do |context| context.assert_not_equals(0.12, 0.12)  end
suite.test 'fail_assert_equals_3' do |context| context.assert_not_equals(true, true)  end
suite.test 'fail_assert_equals_4' do |context| context.assert_not_equals(false, false)  end
suite.test 'fail_assert_equals_5' do |context| context.assert_not_equals('the_string', 'the_' + 'string')  end
suite.test 'fail_assert_equals_6' do |context| context.assert_not_equals({:foo=>1,:bar=>'juu'}, {:foo=>1,:bar=>'juu'})  end
suite.test 'fail_assert_equals_7' do |context| context.assert_not_equals([], [])  end
suite.test 'fail_assert_equals_8' do |context| context.assert_not_equals([123], [123])  end

suite.test 'assert_not_equals_with_message' do |context|
  context.assert_not_equals(0, 1, 'the_message')
  context.assert_not_equals(0.12, 0.13, 'the_message')
  context.assert_not_equals(true, false, 'the_message')
  context.assert_not_equals(false, true, 'the_message')
  context.assert_not_equals('foo', 'bar', 'the_message')
  context.assert_not_equals({:foo=>1,:bar=>'juu'}, {:foo=>1,:bar=>'daa'}, 'the_message')
  context.assert_not_equals([123], [321], 'the_message')
end
suite.test 'fail_assert_equals_with_message_1' do |context| context.assert_not_equals(0, 0, 'the_message')  end
suite.test 'fail_assert_equals_with_message_2' do |context| context.assert_not_equals(0.12, 0.12, 'the_message')  end
suite.test 'fail_assert_equals_with_message_3' do |context| context.assert_not_equals(true, true, 'the_message')  end
suite.test 'fail_assert_equals_with_message_4' do |context| context.assert_not_equals(false, false, 'the_message')  end
suite.test 'fail_assert_equals_with_message_5' do |context| context.assert_not_equals('the_string', 'the_' + 'string', 'the_message')  end
suite.test 'fail_assert_equals_with_message_6' do |context| context.assert_not_equals({:foo=>1,:bar=>'juu'}, {:foo=>1,:bar=>'juu'}, 'the_message')  end
suite.test 'fail_assert_equals_with_message_7' do |context| context.assert_not_equals([], [], 'the_message')  end
suite.test 'fail_assert_equals_with_message_8' do |context| context.assert_not_equals([123], [123], 'the_message')  end

suite.test 'assert_true' do |context| context.assert_true(true)  end
suite.test 'fail_assert_true' do |context| context.assert_true(false)  end

suite.test 'assert_true_with_message' do |context| context.assert_true(true, 'the_message')  end
suite.test 'fail_true_with_message' do |context| context.assert_true(false, 'the_message')  end

suite.test 'assert_false' do |context| context.assert_false(false)  end
suite.test 'fail_assert_false' do |context| context.assert_false(true)  end

suite.test 'assert_false_with_message' do |context| context.assert_false(false, 'the_message')  end
suite.test 'fail_assert_false_with_message' do |context| context.assert_false(true, 'the_message')  end

suite.test 'fail_' do |context|
  context.fail
end

suite.test 'fail_with_message' do |context|
  context.fail 'the_message'
end

suite.run $vertx, { :reporters => [ { :to => 'bus:assert_tests' }] }
